R/tbl_listing.R

```diff
@@ -0,0 +1,129 @@
+#' Data Listing Table
+#'
+#' Function creates a gtsummary-class listing of data. Column labels are
+#' used as column headers, when present.
+#'
+#' @param data a data frame
+#' @param bold_headers logical indicating whether to bold column headers.
+#' Default is `TRUE`
+#' @param group_by Single variable name indicating a grouping variable.
+#' Default is `NULL` for no grouping variable. When specified, a grouping
+#' row will be added to the first column. The grouping column and the first
+#' column in the table will be combined and the type/class may be converted
+#' to common type/class for both columns.
+#'
+#' Both the grouping variable and the
+#' first column must be character.
```

Comment:
Is this a fair description?

```suggestion
#' When the `group_by` argument is utilized, groups are ordered accrording to the grouping 
#'  variable's type (i.e.,  character, numeric, or factor).
```